lib/logsource.c

```diff
@@ -680,6 +691,7 @@ log_source_init_instance(LogSource *self, GlobalConfig *cfg)
   self->super.init = log_source_init;
   self->super.deinit = log_source_deinit;
   window_size_counter_set(&self->window_size, (gsize)-1);
+  self->dynamic_window_initialized = FALSE;
```

Comment:
Since this fields needs to be renamed to `window_initialized`, I think the field could belong to `WindowSizeCounter`.

`window_size_counter_set()` could set this flag, or `window_size_counter_initialize()` can be added instead. What do you think?

----


`window_size_counter_set(&self->window_size, (gsize)-1);` can be removed, the faulty -1 is not used anymore.

Reply:
This field belongs the user, I would not add to the counter struct (and counter represents a counter that can be used in multithreaded environment, while this flag is definitely not)

Reply:
And yes, the extra set should be removed.

Reply:
Making the `release_dynamic_window` call conditionally also a good idea, will update the PR soon. Thanks!